### DISCUSSION TITLE
Fluent Python Journal

### DISCUSSION THREAD

**Discussion by RandSim** [ansys/pyfluent#3285] [Q&A]
Hi,
I am unsuccessful in trying to output a pyFluent journal from Fluent 2024R2. To my demise, the journal file is just a record to GUI actions, not a pyFluent file. This functionality works well with Fluent Mesher but not Solver. Could you point me in the right direction?
Thanks for your help,
Alex,

PS1: I have the python console activated and I am using 2024R2.

PS2: Example journal output is not a pyFluent file
/file/set-tui-version "24.2"
(cx-gui-do cx-activate-item "MenuBar*ReadSubMenu*Mesh...")
(cx-gui-do cx-set-file-dialog-entries "Select File" '( "simple_bend.msh.h5") "CFF Mesh Files (*.msh.h5 )")
(cx-gui-do cx-set-list-tree-selections "NavigationPane*Frame2*Table1*List_Tree2" (list "Setup|Models|Energy (Off)"))
(cx-gui-do cx-set-list-tree-selections "NavigationPane*Frame2*Table1*List_Tree2" (list "Setup|Models|Energy (Off)"))
(cx-gui-do cx-activate-item "NavigationPane*Frame2*Table1*List_Tree2")
(cx-gui-do cx-set-list-tree-selections "NavigationPane*Frame2*Table1*List_Tree2" (list "Setup|Models|Energy (Off)"))
(cx-gui-do cx-set-toggle-button2 "Energy*Table1(Energy)*ToggleBox1*CheckButton1(Energy Equation)" #t)
(cx-gui-do cx-activate-item "Energy*Table1(Energy)*ToggleBox1*CheckButton1(Energy Equation)")
(cx-gui-do cx-activate-item "Energy*PanelButtons*PushButton1(OK)")

**Comment by ansjschuetz** [comment]
Hello RandSim,
what you've found is expected -- only TUI commands can be converted to python code.
For that, when you start the journal file writing, specify a file name that ends in ".pyconsole.py".

I'm not sure about what's been released in 2024 R2, but at least in 2025 R1, there'll be an alternative GUI available by connecting a web browser to the Fluent session. Actions done there will be recorded in the journal file as python code (if you're writing a python journal -- otherwise, WebUI actions will not be journalled at all..).

**Comment by RandSim** [comment]
Jochen,
Thanks for your prompt reply, and great stuff about 2025R1. 
I tried your suggestion using Fluent 2024R2 and started a new journal file called "sim2.pyconsole.py". Content  of the file is shown below. Basically none of my actions seem to be recorded or in this case translated. Am I missing something?

"
import os
if not os.getenv('FLUENT_PROD_DIR'):
    import ansys.fluent.core as pyfluent
    flglobals = pyfluent.setup_for_fluent(product_version="24.2.0", mode="solver", version="3d", precision="double", processor_count=6)
    globals().update(flglobals)
"

**Reply by ansjschuetz** [reply]
Hi RandSim,
OK, at least that looks like the header of a .pyconsole.py file... :-)

What kind of actions did you take that you expected to be recorded into this .pyconsole.py journal file?

Was it TUI commands? If so -- can you try placing them (together with the TUI command to start the .pyconsole.py journal file) in a .jou file and load that into Fluent, e.g. by launching Fluent with the "-i ....jou" command line flag..?

**Reply by RandSim** [reply]
Jochen,

"OK, at least that looks like the header of a .pyconsole.py file... :-)" Yay to small victories!!! :)

I did quite a few things that I was hopping would be recorded. Turning on energy equation, setting density as ideal-gas, create a report, etc.

There was no TUI commands in the .pyconsole.py file but I can get those easily enough with a regular journal file if needed. 

Let me see if I understand your last recommendation correctly. Basically I would launch fluent in batch as opposed to with the GUI, and indicate to my batch fluent session to read in a TUI journal file. At the top of said TUI journal file, I would request to write a journal file ending with .pyconsole.py. If we are successful, this would result in Fluent effectively converting my TUI journal file into a pyFluent journal file. Did I get that right?

**Comment by ansjschuetz** [comment]
Hi RandSim again,

> _I did quite a few things that I was hopping would be recorded._

did you do these _by typing TUI commands_ (like "/define/models/energy ...") into the Fluent text console?
If not, you may want to try that.

> _Did I get that right?_

You did, but I just realise that that won't work, I thought it the wrong way.

Here are modified instructions:
- Prepare a TUI journal (SOMETHING.jou file), including e.g. "/define/models/energy ...".
- At the top of that file, insert a line `(api-start-python-journal "SOMETHING.pyconsole.py")`
- Run Fluent ([as if] in batch mode [could probably omit "-g" to still get the GUI...]), putting "-i SOMETHING.jou" on the command line so that it reads that file and, because of that extra line, converts it into a .pyconsole.py file.

Let me know how that works..

**Comment by mkundu1** [comment]
[Here](https://ansyshelp.ansys.com/public/account/secured?returnurl=/Views/Secured/corp/v242/en/fluent_beta_doc/flu_ug_sec_python_console_interactive.html%23d0e14578) is the documented way of generating Python journal from existing TUI journals.

**Comment by RandSim** [comment]
Thanks Jochen and Mainak! Give me a day or two and Ill give this a shot.
